### ISSUE TITLE
"document.forms is undefined" Error in Firefox when using Saml2PostBinding for Saml2Response

### ISSUE THREAD

**Issue by unsprung** [KentorIT/authservices#202]
The body onload event is called too early in Firefox(tested Versions 10 and 36)  when the Saml2PostBinding response is received causing a "document.forms is undefined" Javascript error. If I copy the html to a static file and then open it in Firefox, it works like it is supposed to.

This appears to be primarily a Firefox problem, but I was able to get around it by overriding Saml2Binding to create a custom PostBinding that gives the html form the id "postform" and substitutes `<body onload=""document.forms[0].submit()"">` for `<body onload=""document.getElementById('postform').submit()"">.`

**Comment by albinsunnanbo** [comment]
Hi
Tanks for the bug report.

http://docs.oasis-open.org/security/saml/v2.0/saml-bindings-2.0-os.pdf notes

> Any technique supported by the user agent MAY be used to cause the submission of the form

Not much of a help with the details.

http://en.wikipedia.org/wiki/SAML_2.0#HTTP_POST_Binding notes

> To automate the submission of the form, the following line of JavaScript may appear anywhere on the XHTML page:
> `  window.onload = function () { document.forms[0].submit(); }`

It looks like our current implementation is very close to the method recommended by the Wikipedia article. Have you done any research suggesting that `document.forms[0]` is problematic in `window.onload` in FireFox?

You are more than welcome to provide a bug fix as a pull request.
See https://github.com/KentorIT/authservices/blob/master/doc/Contributing.md

**Comment by albinsunnanbo** [comment]
I have trouble reproducing.
The only Saml2PostBinding in use that I know of is in the StubIDP, I tried http://stubidp.kentor.se/ with Firefox 36.0 on Windows to one of our own applications with no problem.
Does it always reproduce or is the problem intermittent?
Are there any special tricks to reproduce the bug?

**Comment by unsprung** [comment]
The real problem was that the content-type on the response was missing, which causes Firefox to handle the request differently than other browsers. I should have figured it couldn't have been such a general problem.

I have made my own IDP based on the stub IDP, but it's not running under MVC so I had to adapt the code. With the StubIDP using commandResult.ToActionResult(), the context type appears to be set correctly even though I couldn't find exactly where it's being set so I assume ASP.Net MVC is setting a default somewhere. Anyway, when using commandResult.Apply, the content type is not being set which causes this behavior in FireFox. Perhaps the content type can be defaulted to "text/html" if not already set during Saml2PostBinding.Bind. 

I would like to submit a pull request but I am still a Git noobie and don't have the time right this minute to pick it up.
